contoso_manufacturing/developer/webapp-decode/yolov8.py

```diff
@@ -197,35 +253,10 @@ def draw_fps(self, img):
             cv2.putText(img, label, (label_x, label_y + label_height), font_face, font_scale, font_color, font_thickness, cv2.LINE_AA)
             label_y += (label_height + 2 * pixel_border)
 
-    def run(self):
-        self.log("Running detection...")
-
-        self.frame_number += 1
-        # If mod = 0, i will get the frame and skip it
-        if ((self.skip_rate > 0) and (self.frame_number % self.skip_rate == 0)):
-            self.cap.read()
-            return None
-        
-        # Pre-process the image
-        # Perform inference on the preprocessed image; capture the start and end times
-        # Post-processing including drawing bounding boxes
-        image_data = self.preprocess()
-        time1 = time.time()
-        outputs = self.grpc_client.predict({"images": image_data}, self.model_name)
-        time2 = time.time()
-        frame = self.postprocess(self.cap.read()[1], outputs)
-
-        # Update metrics used for FPS calculations
-        self.total_inference_time += (time2 - time1)
-        self.total_frames += 1
-
-        # Calculate FPS for both the inferencing step and the final feed  
-        self.inference_fps = self.total_frames / self.total_inference_time
-        self.total_fps = self.total_frames / (time.time() - self.start_time)    # This includes e.g. JPEG encoding in the parent method outside of self.run()
-        self.log(f"FPS={self.total_fps} Inference={self.inference_fps:.03f} ({self.total_frames} frames)")
-
-        return frame
-
+    def stop(self):
+        self.stopped = True
+        self.capture_thread.join()
```

Comment:
Should we wait for self.postprocess_thread and self.inference_thread here as well?